### DISCUSSION TITLE
DBeaver not showing explicit column declarations as part of the SQL code in "Definition"

### DISCUSSION THREAD

**Discussion by sk6190** [dbeaver/dbeaver#22036] [Q&A]
When a view is created with explicit column declarations, the view definition does not show that information when viewing through the "View View" -> Definition tab. However, the Columns tab does have the right column names defined through explicit declaration.

So, if I recreate the view or alter it just using the code from the Definition tab, then the information from the explicit declarations are lost. Is there an option to show it correctly in view definition? 

Thanks!

**Comment by LonwoLonwo** [comment]
Hello @sk6190 

What database do you use?

**Reply by sk6190** [reply]
Hi! I'm using SAP HANA.

**Reply by LonwoLonwo** [reply]
We are using this query for HANA

`SELECT DEFINITION FROM SYS.VIEWS WHERE SCHEMA_NAME=? and VIEW_NAME=?`

So, DBeaver shows you the result of this query.

**Reply by sk6190** [reply]
@LonwoLonwo -  Thanks for the response! It doesn't resolve the issue I have though. I assume other tools would have same limitation as well, but the explicit column declarations are included when viewing definition or generating SQL for creating the view again.

**Reply by LonwoLonwo** [reply]
I found the related issue for this: https://github.com/dbeaver/dbeaver/issues/17402

**Reply by sk6190** [reply]
Thanks! Appears to be the same issue and hopefully there will be fix soon for this.

**Reply by LonwoLonwo** [reply]
As I said, we are using definition texts from the system view. And I do not see other options here. Creating a bug report for HANA may resolve this situation.

**Reply by sk6190** [reply]
If column aliases (in explicit column declarations and not within the SELECT)  are used when creating view, then SYS.VIEWS will have the flag HAS_COLUMN_ALIASES set to 'X'. The aliases used can then be read from the SYS.VIEW_COLUMNS for that view. 

In below examples, (1) and (2) view definition in SYS.VIEWS, DEFINTION column is exactly the same. But, the example 2 has column aliases defined and will have flag HAS_COLUMN_ALIASES set to 'X' and aliases will be correctly named in SYS.VIEW_COLUMNS table. 

**Example 1:**
CREATE VIEW ABC.TEST01 
    AS SELECT VEN_ID, TXT FROM VENDOR;

**Example 2:**
CREATE VIEW ABC.TEST02 (VENDOR_ID, VENDOR_NAME) 
    AS SELECT VEN_ID, TXT FROM VENDOR;

Considering other IDEs are able to show this information correctly, I'm not sure how this can be reported as bug to SAP for HANA database.  Below is how the definition is shown in other IDEs. 

**Generated definition for example (1).**
CREATE VIEW ABC.TEST01 (VEN_ID, TXT)
    AS SELECT VEN_ID, TXT FROM VENDOR;

**Generated definition for example (2).**
CREATE VIEW ABC.TEST01 (VENDOR_ID, VENDOR_NAME)
    AS SELECT VEN_ID, TXT FROM VENDOR;
